### DISCUSSION TITLE
How can the user deactivate itself ("is_active": false) without needing the super privileges?

### DISCUSSION THREAD

**Discussion by P-MCC** [fastapi-users/fastapi-users#1252] [Q&A] [answered]
Hello.
I'm new to the fastapi-users library and I hope my question is relevant. 

`patch` users/me route only accepts `username` and `password` in the request body. I am trying to give the request body `is_active` too so that the user can deactivate it's own account. How can it be done?

Any tips will be appreciated &#129303;

**Comment by frankie567** [comment, marked as answer]
Hi @P-MCC 👋 

This is not built-in in the library. However, you can do this pretty easily by implementing your own API. Something like this:

```py
@app.post("/deactivate")
async def deactivate(
    request: Request,
    user: User = Depends(current_active_user),
    user_manager: UserManager = Depends(get_user_manager),
):
    user_update = UserUpdate(is_active=False)
    user = await user_manager.update(user_update, user, safe=True, request=request)
    return {"message": "User deactivated"}
```

**Reply by P-MCC** [reply]
That will do the trick. Thanks a lot @frankie567 &#128515;
